WIP: login portal dashboard

bradewalder opened the pull request:
This is still a work in progress.

Functionality is yet to be added, I have created a basic login portal and dashboard (WIP) based around Ojasvi and I's original Figma designs.

bradewalder committed:
Add dashboard and login components

Introduced new components for the dashboard and login. Still a work in progress. Functionality has not yet been added.

bradewalder committed:
main.tsx

MorgaineBarter committed:
Merge branch 'main' into brade-walder-login-portal-dashboard

MorgaineBarter reviewed (dismissed):
thank you for sharing this on Friday in the meeting. looks good

bradewalder committed:
Added a 'Data & Predictions' page to the dashboard
for data and visualistions to eventually be added.
A drag and drop feature has also been added to read
in data from certain file types. Functionality
can be added to extend this.

bradewalder committed:
Merge branch 'brade-walder-login-portal-dashboard' of https://github.com/bradewalder/redback-fit-web into brade-walder-login-portal-dashboard

bradewalder committed:
Added a 'Data & Predictions' page to the dashboard
for data and visualistions to eventually be added.
A drag and drop feature has also been added to read
in data from certain file types. Functionality
can be added to extend this.

bradewalder committed:
Merge branch 'main' into brade-walder-login-portal-dashboard

doubleedesign commented:
I'll pull and run this code locally ASAP to give more detailed feedback if needed and help get this through, but initial feedback is just to shuffle things around so that each component is in its own folder (inside `src/components`) with its own styling. Some relevant docs published earlier this week:
- [Creating components](https://redback-operations.github.io/redback-documentation/docs/web-mobile-app-dev/frontend/new-components/) (for the expected directory structure and naming conventions)
- [Using Styled Components](https://redback-operations.github.io/redback-documentation/docs/web-mobile-app-dev/frontend/styled-components/)
- [Submitting code](https://redback-operations.github.io/redback-documentation/docs/web-mobile-app-dev/frontend/submitting-work) (more about the linting/structure checks that are now running automatically on PRs to help us keep things consistent)

I'm happy to lend a hand with this if I get time in the next week (but can't promise what time I'll have unfortunately, so don't wait on me).

bradewalder committed:
Tidy up of Structure

bradewalder committed:
Merge branch 'brade-walder-login-portal-dashboard' of https://github.com/bradewalder/redback-fit-web into brade-walder-login-portal-dashboard

doubleedesign commented:
Have set this up locally; loving the animation to switch between login and signup! There could be a few things here to tweak next trimester and maybe some to bring into Redback UI (also next trimester), but overall once we get it passing the checks I'm happy with this as an MVP to be merged.

doubleedesign committed:
Refactor: Put components in their own folders; put page-style ones into Routes; put all routes in main.tsx

doubleedesign committed:
Chore: Get rid of some TypeScript "any"

ben-AI-cybersec commented:
@doubleedesign could you please give us the file naming conventions? The linting errors aren't super clear.

doubleedesign commented:
I've just pushed up a change that corrects the directory structure and naming conventions. I can see that in the future we may want to allow component folders within component folders, but I haven't had a change to tinker with that as yet. So for now at least, the basic formula is:

- routes/
  - RouteName/
    - RouteName.tsx
    - RouteName.style.ts
    - RouteName.test.tsx
- components/
  - ComponentName/
    - ComponentName.tsx
    - ComponentName.style.ts
    - ComponentName.test.tsx

Noting that the file names match the folder names, and all are in `PascalCase` (this is the general React convention). 

I suspect there will also be a `hooks` folder in future, when we come across the need for custom hooks (which will follow the React convention of `camelCase` starting with `use`).

bradewalder reviewed (commented):
Thank you! Looks great!

doubleedesign commented:
No worries @bradewalder! Just the CSS refactoring left now I think, which as discussed on Teams I'm happy to help with ASAP. There's quite a bit here that could potentially be adapted and built upon to be used in other parts of the project and in the other teams' projects too, great work!

doubleedesign commented:
As discussed on Teams, I have raised a PR to allow CSS modules in addition to Styled Components for now, as it will help us to finalise this PR and we can easily convert them later when Redback UI is integrated and we are thus more ready for the dynamic stuff that Styled Components makes available.

bradewalder commented:
Thank you @doubleedesign! I will keep that in mind for the future. Appreciate your insights, they are very valuable.

MorgaineBarter committed:
Merge branch 'main' into brade-walder-login-portal-dashboard

bradewalder committed:
Update / Backend Progress

bradewalder committed:
Update

bradewalder committed:
Dashboard UI Updates

bradewalder committed:
Codeowner change for personal repo

PascalTraczewski committed:
Update package-lock.json

bradewalder committed:
Dashboard UI Updates

PascalTraczewski committed:
Merge branch 'brade-walder-login-portal-dashboard' of https://github.com/bradewalder/redback-fit-web into brade-walder-login-portal-dashboard

PascalTraczewski committed:
update

PascalTraczewski committed:
update 2

bradewalder committed:
Dashboard UI Updates

PascalTraczewski committed:
table

bradewalder committed:
Dashboard UI Update

bradewalder committed:
Dashboard UI Update

PascalTraczewski committed:
c

PascalTraczewski committed:
table

PascalTraczewski committed:
Merge branch 'brade-walder-login-portal-dashboard' into Pascal

PascalTraczewski committed:
Update DashboardLanding.tsx

PascalTraczewski committed:
Merge branch 'brade-walder-login-portal-dashboard' of https://github.com/bradewalder/redback-fit-web into brade-walder-login-portal-dashboard

bradewalder committed:
Merge pull request #1 from PascalTraczewski/brade-walder-login-portal-dashboard

Pascal Changes

bradewalder committed:
Dashboard UI Update / Small Fixes

PascalTraczewski committed:
Merge pull request #1 from bradewalder/brade-walder-login-portal-dashboard

Dashboard UI/Updates

bradewalder committed:
Component Updates

PascalTraczewski committed:
sessionspart

PascalTraczewski committed:
Merge branch 'Redback-Operations:main' into brade-walder-login-portal-dashboard

PascalTraczewski committed:
Merge branch 'pr/2' into brade-walder-login-portal-dashboard

bradewalder committed:
Minor naming changes

bradewalder committed:
Merge pull request #5 from PascalTraczewski/brade-walder-login-portal-dashboard

Brade walder login portal dashboard

bradewalder committed:
Dashboard UI Update

bradewalder committed:
Dashboard UI cleanup / updates

TanayDeakin committed:
Added the HomePage

bradewalder committed:
Merge pull request #7 from TanayDeakin/brade-walder-login-portal-dashboard

Added the HomePage

bradewalder committed:
Frontend tidy up & update

PascalTraczewski committed:
Merge pull request #3 from bradewalder/brade-walder-login-portal-dashboard

Brade walder login portal dashboard

PascalTraczewski committed:
Update package.json

TanayDeakin committed:
Changed the color scheme of the website

bradewalder committed:
Merge pull request #8 from TanayDeakin/brade-walder-login-portal-dashboard

Changed the color scheme of the website

bradewalder committed:
Code review & Update

TanayDeakin committed:
Fixed some errors

bradewalder committed:
Merge pull request #9 from TanayDeakin/brade-walder-login-portal-dashboard

Fixed some errors

PascalTraczewski committed:
perf

PascalTraczewski committed:
perf and hover

PascalTraczewski committed:
Merge branch 'pr/2' into brade-walder-login-portal-dashboard

bradewalder committed:
Merge pull request #11 from PascalTraczewski/brade-walder-login-portal-dashboard

Dashboard Updates

bradewalder committed:
Dashboard CSS Updates

doubleedesign commented on CODEOWNERS:
```diff
@@ -1,5 +1,5 @@
+# @bradewalder will be requested for
 # review when someone opens a pull request.
-*	@Redback-Operations/wearables-for-athletes @Redback-Operations/tutors 
+* @bradewalder @Redback-Operations/wearables-for-athletes @Redback-Operations/tutors 
```
@bradewalder Please remove yourself from the codeowners - I know you're a senior student so you probably shouldn't be an approver once you've finished capstone ;)

bradewalder committed:
CODEOWNERS change

bradewalder committed:
Linting Fix

bradewalder committed:
Linting fix

bradewalder committed:
Linting Fixes & Login Page CSS tweaks

bradewalder committed:
Small CSS fixes + Future notes

bradewalder committed:
Linting Fix

ben-AI-cybersec reviewed (approved):
Moving T2 work from entire team into main repo so we can work through checking it.
